src/java.base/windows/classes/sun/nio/fs/WindowsUserDefinedFileAttributeView.java

```diff
@@ -51,8 +51,19 @@ private String join(String file, String name) {
             throw new NullPointerException("'name' is null");
         return file + ":" + name;
     }
+
     private String join(WindowsPath file, String name) throws WindowsException {
-        return join(file.getPathForWin32Calls(), name);
+        if (name == null)
+            throw new NullPointerException("'name' is null");
+        WindowsFileSystem wfs = file.getFileSystem();
+        WindowsPath namePath = WindowsPath.parse(wfs, name);
+        if (namePath.getRoot() != null)
+            throw new IllegalArgumentException("'name' is not relative");
```

Comment:
The exception message should "has a root component" because a name such as "C:foo" is relative.